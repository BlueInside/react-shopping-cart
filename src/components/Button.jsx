import PropTypes from 'prop-types';

function Button({ label, handleClick }) {
  return <button onClick={handleClick}>{label}</button>;
}

Button.propTypes = {
  label: PropTypes.string,
  handleClick: PropTypes.func.isRequired,
};

Button.defaultProps = {
  label: 'button',
  handleClick: () => console.log('No handleClick prop Provided'),
};

export default Button;
