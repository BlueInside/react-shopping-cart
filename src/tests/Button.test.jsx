import { render, screen } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';
import Button from '../components/Button';
import userEvent from '@testing-library/user-event';

describe('Button component', () => {
  it('renders correctly', () => {
    const { container } = render(<Button />);
    expect(container).toMatchSnapshot();
  });

  it('renders correct label', () => {
    const testLabel = 'test';
    render(<Button label={testLabel} />);
    const button = screen.getByRole('button', { name: testLabel });
    expect(button).toBeInTheDocument();
  });

  it('renders default label "button" if label not provided ', () => {
    render(<Button />);
    const button = screen.getByRole('button', { name: 'button' });
    expect(button).toBeInTheDocument();
  });

  it('calls handleClick function provided in props', async () => {
    const handleClick = vi.fn();
    const user = userEvent.setup();

    render(<Button handleClick={handleClick} />);

    const button = screen.getByRole('button', { name: 'button' });
    await user.click(button);

    expect(handleClick).toHaveBeenCalled();
  });

  it('logs a warning when handleClick is not a function', () => {
    const consoleMock = vi.spyOn(console, 'error');

    render(<Button handleClick={'notAFunction'} />);

    expect(consoleMock).toHaveBeenCalled();

    consoleMock.mockReset();
  });
});
