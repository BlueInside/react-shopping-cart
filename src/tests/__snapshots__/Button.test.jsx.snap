// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button component > renders correctly 1`] = `
<div>
  <button>
    button
  </button>
</div>
`;
